plugins/UFM_NDT_Plugin/ufm_sim_web_service/merger_resources.py

```diff
@@ -381,6 +382,32 @@ def __init__(self):
         self.ndts_dir = self.ndts_merger_dir
         self.reports_dir = self.reports_merger_dir
 
+    def delete_ndt_reports(self, ndt:str):
+        '''
+        Delete reports related to NDT file
+        :param ndt: name of NDT file
+        '''
+        # find reports related to NDT file name and remove them.
+        # Update reports.json file
+        try:
+            with open(self.reports_list_file, "r", encoding="utf-8") as reports_list_file:
+                # unhandled exception in case reports file was changed manually
+                data = json.load(reports_list_file)
+                data_to_proceed = data.copy()
+                for report_data in data_to_proceed:
```

Comment:
this loop can be outside the "with" statement?

Reply:
Possible to open file for read and  read data and then to close. 
With doing the same. In this case both options are valid.

Reply:
you right. it's possible but I thought it will be cleaner...never mind.